src/main/java/mekanism/common/CommonPlayerTickHandler.java

```diff
@@ -124,23 +119,24 @@ private void tickEnd(Player player) {
             }
         }
 
-        ItemStack chest = player.getItemBySlot(EquipmentSlot.CHEST);
-        ItemStack jetpackStack = ItemJetpack.getJetpack(player, chest);
-        JetpackMode mode = getJetpackModeIfOn(player, jetpackStack);
-        if (mode != JetpackMode.DISABLED) {
-            if (handleJetpackMotion(player, mode, () -> Mekanism.keyMap.has(player.getUUID(), KeySync.ASCEND))) {
-                player.fallDistance = 0.0F;
-                if (player instanceof ServerPlayer serverPlayer) {
-                    serverPlayer.connection.aboveGroundTickCount = 0;
+        ItemStack jetpack = IJetpackItem.getActiveJetpack(player);
+        if (!jetpack.isEmpty()) {
+            ItemStack primaryJetpack = IJetpackItem.getPrimaryJetpack(player);
+            if (!primaryJetpack.isEmpty()) {
+                JetpackMode primaryMode = ((IJetpackItem) primaryJetpack.getItem()).getJetpackMode(primaryJetpack);
+                JetpackMode mode = IJetpackItem.getPlayerJetpackMode(player, primaryMode, () -> Mekanism.keyMap.has(player.getUUID(), KeySync.ASCEND));
+                if (mode != JetpackMode.DISABLED) {
+                    // Client handles motion and sets fallDistance in ClientTickHandler at the tick start
+                    if (player instanceof ServerPlayer serverPlayer && IJetpackItem.handleJetpackMotion(player, mode, () -> Mekanism.keyMap.has(player.getUUID(), KeySync.ASCEND))) {
+                        player.fallDistance = 0.0F;
```

Comment:
This should reset fall distance regardless of the player is a server player (even though I am not sure there are any cases that will happen). See the code flow before this PR:
handle -> fall distance -> if server player -> above ground tick count

Reply:
I know I changed it, but client already calls handleJetpackMotion and conditionally resets fall distance every tick in ClientTickHandler. Is there any reason we want to have it call it at tick start and tick end? Or should the ClientTickHandler stuff be moved to common as they're basically doing the same thing. I didn't take it that far as I wasn't sure if it was doing it on tick start for the client intentionally

Reply:
I just meant do this except do the fallDistance = 0 regardless of whether `player instanceof ServerPlayer` in this check. Also this `tickEnd` check only runs on the server side, unlike the client stuff which is on the client side.

Reply:
ok, it's all reverted